### DISCUSSION TITLE
Is there a usage limit?

### DISCUSSION THREAD

**Discussion by chhuang** [wevm/wagmi#2517] [Q&A]
I couldn't find any information regarding the usage limits in the docs. However, I noticed that when using the `useBalance` feature for example, there's a connection established with Ankr, which is a paid service. Who is responsible for covering the costs associated with this connection and how many calls can I make? Just trying to understand how this works.

Thanks!

**Comment by tmm** [comment]
You should provide your own RPC URL. If you don't wagmi falls back to public resources that can be rate-limited.

**Reply by chhuang** [reply]
Thanks for clarifying. Do you have a link where I can read more about this?
